### PR TITLE
refactor/tests: improve coverage

### PR DESCRIPTION
**Pull request by mbruns91**
This is the first PR in a series of PR improving the overal testing in `courier`. This one mainly has the purpose to get a coverage of approx. 100%.

This PR strengthens the unit test suite by covering previously untested branches and adding focused edge-case regression tests around the public client APIs and Ontodocker helpers.

- Add coverage for the installed-package courier.__version__ metadata path
- Add HttpClient tests for empty default_scheme, non-numeric timeout tuples, string-valued verify, and property getters for address, default_scheme, token, session, timeout, and verify
- Add a HttpClient regression test ensuring whitespace-only tokens are normalized away and remove the authorization header
- Add URL-normalization tests for malformed scheme input like http:example.org and URLs with missing host like `https:///path`
- Add `HttpError` tests for dataclass field exposure and string rendering when only a status code is present
- Add Ontodocker compatibility coverage for parsing an empty endpoints list
- Add an Ontodocker dataset test ensuring missing Turtle upload files raise `FileNotFoundError`

Overall, the change is intentionally test-only and tightens confidence around validation, normalization, and
  error-handling behavior without changing runtime code.

### CODE REVIEW

File: tests/unit/test_tests.py [hunk 1]
```diff
@@ -8,3 +10,20 @@ def test_version(self):
...
+            reloaded = importlib.reload(courier)
+
+        self.assertEqual(reloaded.__version__, "1.2.3")
+        importlib.reload(courier)
+
+    def test_version_falls_back_when_package_metadata_is_missing(self):
+        with mock.patch(
+            "importlib.metadata.version",
+            side_effect=importlib.metadata.PackageNotFoundError,
+        ):
```
[the remark this answers is not in the document]
Comment (mbruns91) [reply tests/unit/test_tests.py:26]: Fixed via [b567400](https://github.com/pyiron/courier/pull/43/commits/b567400eee88dcd372dccde5c0e00d34cd0dd2c9): directly do the import: `from importlib.metadata import PackageNotFoundError`

File: tests/unit/test_ontodocker_client.py [hunk 2]
```diff
@@ -215,6 +215,13 @@ def test_upload_turtlefile_validates_inputs(self):
         with self.assertRaises(ValidationError):
             _ = c.datasets.upload_turtlefile("ds", "")
 
+    def test_upload_turtlefile_missing_file_raises_file_not_found(self):
+        s = _FakeSession()
+        c = OntodockerClient("https://example.org", session=s)
+
+        with self.assertRaises(FileNotFoundError):
+            _ = c.datasets.upload_turtlefile("ds", "/nonexistent/path.ttl")
```
[the remark this answers is not in the document]
Comment (mbruns91) [reply tests/unit/test_ontodocker_client.py]: Solved via [ab4ad72](https://github.com/pyiron/courier/pull/43/commits/ab4ad724bda498b50a55c7cbf1aabcf3c209ae19):  Replaced the hard-coded absolute missing path with a guaranteed-missing file inside a fresh `TemporaryDirectory`, so the `FileNotFoundError` assertion is deterministic (basically, copilot's suggestion).

**Comment by mbruns91** [comment]
Merging this PR solves some parts of #39.

**Comment by samwaseda** [comment]
@copilot Make sure that the correct `InvalidAddressError` is raised in `normalize_base_url` in the unit tests by checking the error message content

**Review by samwaseda — APPROVED** [review]
